kano_updater/commands/download.py

```diff
@@ -55,10 +55,29 @@ def download(progress=None, gui=True):
         progress.start('downloading')
 
     elif status.state == UpdaterStatus.UPDATES_DOWNLOADED:
-        err_msg = N_("Updates have been downloaded already")
-        logger.error(err_msg)
-        progress.abort(_(err_msg))
-        return True
+        progress.split(
+            Phase(
+                'checking',
+                _("Checking for updates"),
+                10,
+                is_main=True
+            ),
+            Phase(
+                'downloading',
+                _("Downloading updates"),
+                90,
+                is_main=True
+            )
+        )
+        progress.start('checking')
+        check_for_updates(progress=progress)
+        if status.state == UpdaterStatus.NO_UPDATES:
+            err_msg = N_("Latest updates have been downloaded already")
+            logger.info(err_msg)
+            progress.abort(_(err_msg))
```

Comment:
Hmm, I'm not sure about this `abort()` anymore. I think that we were aborting before because this is the message that you would get if you ran the check for updates in GUI mode and there now being no updates. Admittedly, the only way that I can see of getting to this situation is if you've downloaded the updates and then for some reason we've reverted our repo to a previous version but I'm not exactly certain of the behaviour in this case.

Reply:
I've had a rethink and went through the code again. I believe it is what we'd want to do in that case - show an error message and quit the app. If you've gotten here by switching the sources, then it would effectively report there isn't an update available where you switched. The background mode would revert the state to a NO_UPDATES and the GUI mode would show no updates and revert the state again. Both would should get rid of messages probing the user to update 🤔